extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/client/VertxVaultClient.java

```diff
@@ -377,143 +386,147 @@ public void deletePolicy(String token, String name) {
-        final Request request = builder(path, queryParams).get().build();
+        final HttpRequest<Buffer> request = builder(path, queryParams).method(HttpMethod.GET);
         return exec(request, resultClass);
     }
 
     protected int head(String path) {
-        final Request request = builder(path).head().build();
+        final HttpRequest<Buffer> request = builder(path).method(HttpMethod.HEAD);
         return exec(request);
     }
 
     protected int head(String path, Map<String, String> queryParams) {
-        final Request request = builder(path, queryParams).head().build();
+        final HttpRequest<Buffer> request = builder(path, queryParams).method(HttpMethod.HEAD);
         return exec(request);
     }
 
-    private <T> T exec(Request request, Class<T> resultClass) {
-        return exec(request, resultClass, 200);
+    private <T> T exec(HttpRequest<Buffer> request, Class<T> resultClass) {
+        return exec(request, null, resultClass, 200);
     }
 
-    private <T> T exec(Request request, int expectedCode) {
-        return exec(request, null, expectedCode);
+    private <T> T exec(HttpRequest<Buffer> request, int expectedCode) {
+        return exec(request, null, null, expectedCode);
     }
 
-    private <T> T exec(Request request, Class<T> resultClass, int expectedCode) {
-        try (Response response = client.newCall(request).execute()) {
-            if (response.code() != expectedCode) {
+    private <T> T exec(HttpRequest<Buffer> request, Object body, Class<T> resultClass) {
+        return exec(request, body, resultClass, 200);
+    }
+
+    private <T> T exec(HttpRequest<Buffer> request, Object body, Class<T> resultClass, int expectedCode) {
+        try {
+            HttpResponse<Buffer> response = body == null
+                    ? request.sendAndAwait()
+                    : request.sendBufferAndAwait(Buffer.buffer(requestBody(body)));
+            if (response.statusCode() != expectedCode) {
                 throwVaultException(response);
             }
-            String jsonBody = response.body().string();
-            return resultClass == null ? null : mapper.readValue(jsonBody, resultClass);
-        } catch (IOException e) {
+            Buffer responseBuffer = response.body();
+            if (responseBuffer != null) {
+                return resultClass == null ? null : mapper.readValue(responseBuffer.toString(), resultClass);
+            } else {
+                return null;
+            }
+        } catch (JsonProcessingException e) {
             throw new VaultException(e);
         }
     }
 
-    private int exec(Request request) {
-        try (Response response = client.newCall(request).execute()) {
-            return response.code();
-        } catch (IOException e) {
-            throw new VaultException(e);
-        }
+    private int exec(HttpRequest<Buffer> request) {
+        return request.sendAndAwait().statusCode();
```

Comment:
Same issue here, I would add a timeout, just in case...

Reply:
done. I reused the read timeout, which is part of the config.